src/shogun/regression/KRRNystrom.h

```diff
@@ -0,0 +1,134 @@
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ *
+ * The views and conclusions contained in the software and documentation are those
+ * of the authors and should not be interpreted as representing official policies,
+ * either expressed or implied, of the Shogun Development Team.
+ */
+
+#ifndef _KRRNYSTROM_H__
+#define _KRRNYSTROM_H__
+
+#include <shogun/regression/KernelRidgeRegression.h>
+
+namespace shogun {
+
+/** @brief Class KRRNystrom implements the Nyström method for kernel ridge
+ * regression, using a low-rank approximation to the kernel matrix.
+ *
+ * The method is equivalent to ordinary kernel ridge regression, but through a
+ * low-rank approximation to the original kernel matrix, the full kernel matrix
+ * does not need to be computed, and the resulting system of equations for the
+ * alphas is cheaper to solve.
+ *
+ * Given a data matrix X with n training examples along the rows, one chooses
+ * a subset m of the training example and projects all training examples upon
+ * these. This yields the following approximation to the kernel matrix
+ *
+ * \f[
+ * K \approx K_{n,m}K^+_{m,m}K_{m,n}
+ * ]
+ * where \f$K_{n,m}\f$ is a submatrix containing all n rows and the m columns
+ * corresponding to the m chosen training examples, \f$K_{m,n}\f$ is its
+ * transpose and \f$K_{m,m} is the submatrix with the m rows and columns
+ * corresponding to the training examples chosen. \f$+\f$ indicates the
+ * Moore-Penrose pseudoinverse.
+ *
+ * The resulting system of equations for the alphas can then be solved through
+ * the Woodbury formula, which gives the following equation for the alphas
+ *
+ * \f[
+ * {\bf \alpha} = \frac{1}{\tau} ({\bf y} - K_{n,m} (\tau K_{m,m} + K_{m,n}K_{n,m})^+K_{m,n} {\bf y}
+ * ]
+ */
+class CKRRNystrom : public CKernelRidgeRegression
+{
+public:
+	MACHINE_PROBLEM_TYPE(PT_REGRESSION);
+
+	/** Default constructor */
+	CKRRNystrom();
+
+	/** Constructor
+	 *
+	 * @param tau regularization parameter tau
+	 * @param m number of rows/columns to choose
+	 * @param k kernel
+	 * @param lab labels
+	 */
+	CKRRNystrom(float64_t tau, int32_t m, CKernel* k, CLabels* lab);
+
+	/** Default destructor */
+	virtual ~CKRRNystrom() {}
+
+	/** Set the number of columns/rows to choose
+	 *
+	 * @param m new m
+	 */
+	inline void set_m(int32_t m) { m_m = m; };
+
+	/** Set the regularization parameter
+	 *
+	 * @param tau new tau
+	 */
+	inline void set_tau(float64_t tau) { m_tau = tau; };
+
+	/** @return object name */
+	virtual const char* get_name() const { return "KRRNystrom"; }
+
+	/**
+	 * Get the classifier type
+	 *
+	 * @return classifier type KernelRidgeRegression
+	 */
+	EMachineType get_classifier_type()
+	{
+		return CT_KERNELRIDGEREGRESSION;
+	}
+
+protected:
+	/** Train regression using the Nyström method.
+	 *
+	 * @return boolean to indicate success
+	 */
+	bool solve_krr_system();
```

Comment:
Maybe also put the keyword here, so that I wont moan like 10 more times because I forgot you changed the base class ;)